server/src/main/java/org/opensearch/cluster/SnapshotsInProgress.java

```diff
@@ -453,6 +454,29 @@ public Entry withClones(final Map<RepositoryShardId, ShardSnapshotStatus> update
             );
         }
 
+        public Entry withRemoteStoreIndexShallowCopy(final boolean remoteStoreIndexShallowCopy) {
+            if (remoteStoreIndexShallowCopy == this.remoteStoreIndexShallowCopy) {
```

Comment:
how does `this.remoteStoreIndexShallowCopy` matter here ? It should all depend upon the source snapshot . If that shallow , this is going to be shallow .